internal/topology/variables/cluster_variable_defaulting.go

```diff
@@ -83,6 +83,24 @@ func defaultClusterVariable(clusterVariable *clusterv1.ClusterVariable, clusterC
 			fmt.Sprintf("invalid schema in ClusterClass for variable %q: error to convert schema %v", clusterVariable.Name, err))}
 	}
 
+	var value interface{}
+	if len(clusterVariable.Value.Raw) > 0 {
+		if err := json.Unmarshal(clusterVariable.Value.Raw, &value); err != nil {
+			return field.ErrorList{field.Invalid(fldPath, "",
+				fmt.Sprintf("could not unmarshal variable value %q: %v", string(clusterVariable.Value.Raw), err))}
+		}
+	}
+	// If the value is nil and the schema has a default value, default it.
+	// NOTE: This is essentially the same which the structural schema defaulting
+	// lib would do, except that it wouldn't default nullable values.
+	// As we want to also default nullable variables we have to do it ourselves.
+	if value == nil && apiExtensionsSchema.Default != nil {
```

Comment:
Please note:
For basic types those l.97-l.99 could essentially replace everything from l.86 until l.122.

But I would really like to keep the rest of the code as otherwise I have to reintroduce it with the support for object variables and arrays.